content/articles/automating-jobs-scheduling-with-django_cron-in-python-application/index.md

```diff
@@ -0,0 +1,394 @@
+        result.append(data)
+
+    return result
+
+
+# get the latest hackernews streamed
+  def get(self, request, format=None):
+    return Response(self.get_data_from_API(),status=status.HTTP_201_CREATED)
+```
+
+
+#### Configuring routes for your endpoint
+Under the project directory, open to the `urls.py` and paste the snippet below. These are the routes for the `rest-framework`, `admin`, and the `news` app made.
+
+```python
+from django.contrib import admin
+from django.urls import path, include
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    path('api-auth/', include('rest_framework.urls')),
+    path('api/v0/items/', include('news.urls'))
+]
+```
+
+To connect the `news` app with the `hackernewsdemo` project route, you will need to make your custom `urls.py` inside the app directory. Then add the code below and paste it in.
+
+```python
+from django.urls import path
+from .views import NewsIdView, NewsItemView
+
+urlpatterns = [
+    path('all/', NewsIdView.as_view(), name='index'),
+    # http://127.0.0.1:8080/api/v0/items/all --> lists all news id from hackernews endpoint
+    path('hackernews/', NewsItemView.as_view(), name='news-item'), 
+    # http://127.0.0.1:8080/api/v0/items/hackernews --> GET the latest hackernews from db
+]
+```
+
+#### Creating an admin site
+As a Django developer, you are familiar with the default admin page that can be configured. Executing the command below will make a brand new administration site where all activities of the database are controlled.
+
+```bash
+python manage.py createsupuer
+```
+
+Having created a superuser admin account, you will have to register the `news` app model in the `admin.py`. Add this code in the `admin.py` file.
+
+```python
+from django.contrib import admin
+from .models import HackerNewsID, QuickCheckItem, QuickCheckNews
+
+
+class HackerNewsIDAdmin(admin.ModelAdmin):
+    list_display = ('hackernews', 'fetched_at') #these are the features to be listed
+    list_display_links = ('hackernews', 'fetched_at') #these are the features links
+
+admin.site.register(HackerNewsID, HackerNewsIDAdmin)
+```
+
+Furthermore, you can test to the admin page made by executing the following command. 
+
+```bash
+python manage.py runserver
+```
+
+- Note: Now open <http://127.0.0.1:8000/admin> in your browser and log in.
+
+Running the command showcased below will perform the API requests to the external enpoints from the cron class. This is to cycle through the endpoints inside the `cron.py` job class. By doing this, the latest `ids` generated wil be added to the application database.
+
+
+```bash
+python manage.py runcrons
+```
+
+![Admin page](/engineering-education/automating-jobs-scheduling-with-django_cron-in-python-application/admin.png)
+
+In addition, you can make an API request to the endpoint <http://127.0.0.1:8000/api/v0/items/hackernews> to fetch the data that each of the saved ids return. You should get something like shown in the image below.
```

Comment:
The link: http://127.0.0.1:8000/api/v0/items/hackernews throws the following error:

```
AssertionError at /api/v0/items/hackernews/

'NewsItemView' should either include a `queryset` attribute, or override the `get_queryset()` method.
```

Please fix this.

Reply:
Fixed.